keps/sig-node/3759-sidecar-containers/README.md

```diff
@@ -0,0 +1,1282 @@
+  sidecar to initialize should be the last sidecar to terminate.
+
+To address these needs, before promoting this enhancement to Beta, we introduce
+additional termination grace period fields to manage termination duration
+([draft proposal](https://docs.google.com/document/d/1B01EdgWJAfkT3l6CIwNwoskQ71eyuwet3mjiQrMQbU8))
+and leverage these fields to add reverse order termination of sidecar containers
+after the primary containers terminate.
+
+### Other
+
+This behavior needs to be adjusted:
+https://github.com/kubernetes/enhancements/issues/3676 
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+##### Pod lifecycle scenarios without sidecar containers
+
+TBD: describe test cases that can be affected by introducing sidecar containers.
+
+##### Pod lifecycle scenarios with sidecar containers
+
+TBD: describe test cases to test with sidecar containers.
```

Comment:
I saw this section is TBD, but so it won't get missed, I think sidecar containers will affect LimitRange admission.  Otherwise a sidecar could be used to bypass a LimitRange.

Reply:
added